compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -241,10 +314,10 @@ private class ExtractDependenciesCollector(implicit val ctx: Context) extends tp
     sym.isAnonymousFunction ||
     sym.isAnonymousClass
 
-  private def addInheritanceDependency(sym: Symbol): Unit =
-    _topLevelInheritanceDependencies += sym.topLevelClass
+  private def addInheritanceDependency(sym: Symbol)(implicit ctx: Context): Unit =
+    _topLevelInheritanceDependencies += ((currentClass, sym.topLevelClass))
 
-  private object PatMatDependencyTraverser extends ExtractTypesCollector {
+  private class PatMatDependencyTraverser(ctx0: Context) extends ExtractTypesCollector(ctx0) {
```

Comment:
Why not `implicit ctx: Context` ? And at the same time, the `implicit ctx: Context` argument to method in this class should be removed.